heat/core/manipulations.py

```diff
@@ -630,6 +637,79 @@ def diagonal(a, offset=0, dim1=0, dim2=1):
     return factories.array(result, dtype=a.dtype, is_split=split, device=a.device, comm=a.comm)
 
 
+def dsplit(ary, indices_or_sections):
+    """
+    Split array into multiple sub-arrays along the 3rd axis (depth).
+
+    Parameters
+    ----------
+    ary : DNDarray
+        DNDArray to be divided into sub-DNDarrays.
+    indices_or_sections : int or 1-dimensional array_like (i.e. undistributed DNDarray, list or tuple)
+        If indices_or_sections is an integer, N, the DNDarray will be divided into N equal DNDarrays along the 3rd axis.
+        If such a split is not possible, an error is raised.
+        If indices_or_sections is a 1-D DNDarray of sorted integers, the entries indicate where along the 3rd axis
+        the array is split.
+        If an index exceeds the dimension of the array along the 3rd axis, an empty sub-array is returned correspondingly.
+
+    Returns
+    -------
+    sub_arrays : list of DNDarrays
+        A list of sub-DNDarrays as views into ary.
+
+    Notes
+    -----
+    Please refer to the split documentation. dsplit is equivalent to split with axis=2,
+    the array is always split along the third axis provided the array dimension is greater than or equal to 3.
+
+    Raises
+    ------
+    ValueError
+        If indices_or_sections is given as integer, but a split does not result in equal division.
+
+    See Also
+    ------
+    :function:`split`
+
+    Examples
+    --------
+    >>> x = ht.array(24).reshape((2, 3, 4))
+    >>> ht.dsplit(x, 2)
+        [
+            DNDarray([[[ 0,  1],
+                       [ 4,  5],
+                       [ 8,  9]],
+                       [[12, 13],
+                       [16, 17],
+                       [20, 21]]]),
+            DNDarray([[[ 2,  3],
+                       [ 6,  7],
+                       [10, 11]],
+                       [[14, 15],
+                       [18, 19],
+                       [22, 23]]])
+        ]
+    >>> ht.dsplit(x, [1, 4])
+        [
+            DNDarray([[[ 0],
+                        [ 4],
+                        [ 8]],
+                       [[12],
+                        [16],
+                        [20]]]),
+            DNDarray([[[ 1,  2,  3],
+                        [ 5,  6,  7],
+                        [ 9, 10, 11]],
+                        [[13, 14, 15],
+                         [17, 18, 19],
+                         [21, 22, 23]]]),
+            DNDarray([])
+        ]
+
+    """
```

Comment:
here we should raise an exception if `ary.ndim < 3`

Reply:
If that is the case, an exception is raised within `split`, so I think raising an additional one in `dsplit` would be superfluous